<launch>

  <arg name="launch_right" default="true" />
  <arg name="launch_left" default="true" />

  <!-- right gripper -->
  <group ns="gripper_front/limb/right/dxl" if="$(arg launch_right)">
    <!-- Start the Dynamixel servo controller manager -->
    <node name="controller_manager" pkg="dynamixel_controllers" type="controller_manager.py" respawn="true" output="screen"
          machine="c2">
      <rosparam>
        namespace: dxl_manager
        serial_ports:
          port:
            port_name: "/dev/r_dxhub"
            baud_rate: 57600
            min_motor_id: 1
            max_motor_id: 4
            update_rate: 20
      </rosparam>
    </node>

    <!-- Start the Dynamixel servo controllers -->
    <rosparam file="$(find jsk_arc2017_baxter)/robots/right_gripper_v6/dxl_controllers.yaml" command="load"/>
    <node name="controller_spawner"
          pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port port
                prismatic_joint_controller
                vacuum_pad_tendon_controller
                finger_yaw_joint_controller
                finger_tendon_controller"
          output="screen"
          machine="c2" />
    <node name="check_gripper_state"
          pkg="jsk_arc2017_baxter" type="check-gripper-state.l"
          output="screen" respawn="true"
          machine="c2">
      <param name="target_node_name" value="/gripper_front/limb/right/dxl/controller_manager" />
    </node>

  </group>

  <!-- left gripper -->
  <group ns="gripper_front/limb/left/dxl" if="$(arg launch_left)">
    <!-- Start the Dynamixel servo controller manager -->
    <node name="controller_manager"
          pkg="dynamixel_controllers" type="controller_manager.py"
          respawn="true" output="screen"
          machine="c2">
      <rosparam>
        namespace: dxl_manager
        serial_ports:
          port:
            port_name: "/dev/l_dxhub"
            baud_rate: 57600
            min_motor_id: 1
            max_motor_id: 4
            update_rate: 20
      </rosparam>
    </node>

    <!-- Start the Dynamixel servo controllers -->
    <rosparam file="$(find jsk_arc2017_baxter)/robots/left_gripper_v6/dxl_controllers.yaml" command="load"/>
    <node name="controller_spawner"
          pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port port
                prismatic_joint_controller
                vacuum_pad_tendon_controller
                finger_yaw_joint_controller
                finger_tendon_controller"
          output="screen"
          machine="c2" />
    <node name="check_gripper_state"
          pkg="jsk_arc2017_baxter" type="check-gripper-state.l"
          output="screen" respawn="true"
          machine="c2">
      <param name="target_node_name" value="/gripper_front/limb/left/dxl/controller_manager" />
    </node>

  </group>

</launch>
